src/components/TodoCreationForm.js

```diff
@@ -1,27 +1,48 @@
 import React from "react";
+import DatePicker from "react-datepicker";
+import Button from "./library/Button";
 
 export default class TodoCreationForm extends React.Component {
   state = {
-    taskName: ""
+    todoName: "",
+    dueDate: null
   }
 
   render() {
     return (
       <form
+        className="todo-creation-form"
         onSubmit={e => {
           e.preventDefault();
-          if (this.state.taskName === "") return;
-          this.props.onCreate(this.state.taskName)
-          this.setState({taskName: ""})
+          if (this.state.todoName === "") return;
```

Comment:
Here too, you need to return unless both fields, todoName and dueDate are filled. So the condition that you use to disable the button should also appear here. That is why I suggested that you define a `canSubmit` variable that you would then use both in the form and in the button

Reply:
aah ok got it